pkg/providers/pricing/pricing.go

```diff
@@ -405,6 +405,7 @@ func populateInitialSpotPricing(pricing map[string]float64) map[string]zonal {
 }
 
 func (p *Provider) Reset() {
+	initialOnDemandPrices := lo.Assign(initialOnDemandPricesUS, initialOnDemandPricesUS)
```

Comment:
Store this in a top-level var? Then you don't have to get this from a local value, you just always respect the global.

Reply:
I could go either way, if it was here alone for sure. It's also used [here](https://github.com/aws/karpenter/blob/2758afa7fe86b40475193c3322ef70fd0a1c391f/pkg/providers/pricing/pricing.go#L152) though so I separated it out since a ternary seemed a little messy there.

Reply:
Advantage is that there's no reason to scope it. I don't think you're polluting the global context by doing this. This really _should_ be a global since it's just combining two other globals

Reply:
You sold me on lifting it to global scope, but idk if its really possible with go's package init order. Since globals in the same package are initialized in ascending alphabetical order by their file name, the generated files are always initialized last. We could continue initializing the var in `Reset`, but at that point I'd rather just keep it as a local.